[Microsoft.StorageSync] Support Private Endpoint Connection APIs

ankushbindlish2 opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).
Inspired from Other RP PR
https://github.com/Azure/azure-rest-api-specs/pull/7631
https://github.com/Azure/azure-rest-api-specs/pull/7440

ankushbindlish2 committed:
Add Private Link Methods

AutorestCI commented:
/azp run automation - sdk

ankushbindlish2 committed:
fix example

AutorestCI commented:
/azp run automation - sdk

ankushbindlish2 committed:
Update storagesync.json

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?

ankushbindlish2 committed:
Update StorageSyncServices_Get.json

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

kszobi commented on specification/storagesync/resource-manager/Microsoft.StorageSync/preview/2020-03-01/examples/StorageSyncServices_Get.json:
```diff
@@ -1,28 +1,41 @@
+                        "properties": {
+                            "privateEndpoint": {
+                                "id": "subscriptions/52b8da2f-61e0-4a1f-8dde-336911f367fb/resourceGroups/SampleResourceGroup_2/providers/Microsoft.Network/privateEndpoints/testpe01"
+                            },
```
Is privateEndpointConnections part of every GetStorageSyncService GET operaration now? Is this want we want/need? This will negatively impact performance. I'm wondering why are they not their own "resource".

kszobi commented on specification/storagesync/resource-manager/Microsoft.StorageSync/preview/2020-03-01/examples/PrivateLinkResources_List.json:
```diff
@@ -0,0 +1,33 @@
+            "properties": {
+              "groupId": "afs",
+              "requiredMembers": [
+                "mgmt",
```
management

ankushbindlish2 committed:
Add swagger changes for Namespace and FileSyncDownload Policy

AutorestCI commented:
/azp run automation - sdk

ankushbindlish2 committed:
Update PrivateLinkResources_List.json

AutorestCI commented:
/azp run automation - sdk

Manishd79 reviewed (approved):
Changes for the two new properties looks good.

akning-ms commented:
HI @ankushbindlish2  this is some validation error detected. e.g. Semantic and Modal Validation and PretterCheck, can you fix it in this order?

ankushbindlish2 committed:
Update storagesync.json

AutorestCI commented:
/azp run automation - sdk

ankushbindlish2 committed:
Update storagesync.json

AutorestCI commented:
/azp run automation - sdk

ankushbindlish2 committed:
Update storagesync.json

AutorestCI commented:
/azp run automation - sdk

ankushbindlish2 committed:
Update

AutorestCI commented:
/azp run automation - sdk

ankushbindlish2 committed:
Update storagesync.json

AutorestCI commented:
/azp run automation - sdk

ankushbindlish2 committed:
Update storagesync.json

AutorestCI commented:
/azp run automation - sdk

akning-ms commented:
/azp run automation - sdk

ankushbindlish2 committed:
Prettier check

h:\SourceCode\azure-rest-api-specs\specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01>npm run prettier -- --write "H:\SourceCode\azure-rest-api-specs\specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\storagesync.json"

> azure-rest-api-specs-tests@0.1.0 prettier h:\SourceCode\azure-rest-api-specs
> prettier "--write" "H:\SourceCode\azure-rest-api-specs\specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\storagesync.json"

specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\storagesync.json 240ms

h:\SourceCode\azure-rest-api-specs\specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01>npm run prettier -- --write "H:\SourceCode\azure-rest-api-specs\specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\**/*.json"

> azure-rest-api-specs-tests@0.1.0 prettier h:\SourceCode\azure-rest-api-specs
> prettier "--write" "H:\SourceCode\azure-rest-api-specs\specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\**/*.json"

specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_Create.json 16ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_Delete.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_Get.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_ListBySyncGrspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_ListBySyncGroup.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_PostBackup.jspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_PostBackup.json 4ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_PostRestore.specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_PostRestore.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_PreBackup.jsspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_PreBackup.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_PreRestore.jspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_PreRestore.json 6ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_RestoreHeatbspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_RestoreHeatbeat.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_TriggerChangspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\CloudEndpoints_TriggerChangeDetection.json 6ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\Operations_List.json 12ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\OperationStatus_Get.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateEndpointConnections_specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateEndpointConnections_Create.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateEndpointConnections_specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateEndpointConnections_Delete.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateEndpointConnections_specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateEndpointConnections_Get.json 4ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateEndpointConnections_specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateEndpointConnections_ListByStorageSyncService.json 6ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateLinkResources_List.jspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\PrivateLinkResources_List.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_Create.jsspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_Create.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_Delete.jsspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_Delete.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_Get.json 12ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_ListByStospecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_ListByStorageSyncService.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_TriggerRospecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\RegisteredServers_TriggerRollover.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\ServerEndpoints_Create.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\ServerEndpoints_Delete.json 6ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\ServerEndpoints_Get.json 10ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\ServerEndpoints_ListBySyncGspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\ServerEndpoints_ListBySyncGroup.json 17ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\ServerEndpoints_Recall.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\ServerEndpoints_Update.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServiceCheckNamespecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServiceCheckNameAvailability_AlreadyExists.json 3ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServiceCheckNamespecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServiceCheckNameAvailability_Available.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_Create.specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_Create.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_Delete.specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_Delete.json 6ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_Get.jsospecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_Get.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_ListByRspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_ListByResourceGroup.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_ListBySspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_ListBySubscription.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_Update.specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\StorageSyncServices_Update.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\SyncGroups_Create.json 6ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\SyncGroups_Delete.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\SyncGroups_Get.json 8ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\SyncGroups_ListByStorageSynspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\SyncGroups_ListByStorageSyncService.json 4ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\Workflows_Abort.json 4ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\Workflows_Get.json 5ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\Workflows_ListByStorageSyncspecification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\examples\Workflows_ListByStorageSyncService.json 7ms
specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01\storagesync.json 167ms

h:\SourceCode\azure-rest-api-specs\specification\storagesync\resource-manager\Microsoft.StorageSync\preview\2020-03-01>

AutorestCI commented:
/azp run automation - sdk

akning-ms commented:
According to discuss with ankushbindlish2 offline, will request ARM review later when merge into master
